utils/python/CIME/system_test.py

```diff
@@ -243,8 +243,15 @@ def _log_output(self, test, output):
     ###########################################################################
     def _get_case_id(self, test):
     ###########################################################################
-        baseline_action_code = ".C" if self._compare else (".G" if self._generate else "")
-        return "%s%s.%s" % (test, baseline_action_code, self._test_id)
+        baseline_action_code = ""
```

Comment:
cesm allows both compare and generate flags, this should be reflected in testname